grails-app/conf/BootStrap.groovy

```diff
@@ -104,7 +104,8 @@ class BootStrap {
             [
                     id         : shipmentType.id,
                     name       : shipmentType.name,
-                    description: shipmentType.description
+                    description: shipmentType.description,
+                    label      : shipmentType.displayName,
```

Comment:
Added this to DTO to avoid some costly, complexed logic on the frontend side, which hasn't even't worked in 100% like it should due to async conflicts happening.

Reply:
To me currently this looks like a good solution but still maybe requires a tech huddle discussion since we are dealing with translating values from the database.
My general question is why was the **shipmentType** a DB record value instead of an enum to begin with?
Correct me if I am wrong but currently I don't see a workflow that would allow us to add new or delete shipment types so maybe we should look into a potential refactor to follow our convention of defining static types/values as enums, that way we can just follow our method of translation using crowdin like for every other type.

Reply:
yeah, I agree it probably shouldn't have been created as a DB table, but maybe there was some argument for that to be a table instead of an enum. 
I think this is only it, that is not an enum (like statuses etc), but the table. I think that this refactor should be maybe handled after grails migration, since there is a lot of associations between a domain and `ShipmentType` table, so it might be an overkill to handle it now, but I agree with you.

Reply:
I also think that another approach could be to just store a translation code having an additional column in the `ShipmentType` table, but I feel like my solution is more "dynamic", so whenever we decide to change any prefix or anything in the translation code, we do not have to do migrations etc. But instead of even doing so, we should probably in the future think about doing what you've said - migrating it to an enum

Reply:
> My general question is why was the shipmentType a DB record value instead of an enum to begin with?

Because it predated our convention of using type table to wrap enums. The shipment domains (Shipment, ShipmentType, and ShipmentItem) were literally the first set of domains after Product and InventoryItem and ShipmentType was likely the first "type" domain. It hasn't been upgraded because we haven't made any significant changes to the Shipment data model since we started.

We should definitely refactor ShipmentType to include the enum, but to also incorporate a more complex real-world data model (i.e. carriers, shipping methods, etc) around it. The user should not ever be selecting a shipment type of Air, Sea, etc. They should be choosing a specific shipping option available from a carrier UPS Ground, FedEx Priority Overnight®, DHL International Direct. And those shipping options would be tied to a shipment type. 

However, it's not always that simple since it's likely (for both international and domestic shipments) there will be multiple modes of transportation involved i.e. sea container received at port of entry, truck picks up container and delivers to customs clearance dropoff, another truck picks up container and drives to train depot to be loaded onto a train. 

You get the idea. The current shipment type domain is not very useful as it is currently constituted so this needs more discussion. The good news is that PIH has no interest in changing it so this might be something I need to do on my own.